<!--
  This file is part of the uav_landing_sim distribution (https://github.com/dimianx/uav_landing_sim).
  Copyright (c) 2024-2025 Dmitry Anikin <dmitry.anikin@proton.me>.

  This program is free software: you can redistribute it and/or modify  
  it under the terms of the GNU General Public License as published by  
  the Free Software Foundation, version 3.

  This program is distributed in the hope that it will be useful, but 
  WITHOUT ANY WARRANTY; without even the implied warranty of 
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU 
  General Public License for more details.

  You should have received a copy of the GNU General Public License 
  along with this program. If not, see <http://www.gnu.org/licenses/>.
-->


<launch>
    <arg name="attempts_count" default="1"/>
    <arg name="output" default="results.csv"/>

    <node name="drone_fsm" pkg="uav_landing_sim" type="drone_fsm.py" respawn="false" output="screen">
        <rosparam file="$(find uav_landing_sim)/config/drone/drone_fsm.yaml" command="load"/>
        <param name="attempts_count" type="int" value="$(arg attempts_count)" />
        <param name="output" type="str" value="$(arg output)" />
    </node>

</launch>